modules/dnn/src/layers/harmax.cpp

```diff
@@ -0,0 +1,151 @@
+        return false;
+    }
+
+    void forward(InputArrayOfArrays inputs_arr,
+                 OutputArrayOfArrays outputs_arr,
+                 OutputArrayOfArrays internals_arr) CV_OVERRIDE
+    {
+        CV_TRACE_FUNCTION();
+        CV_TRACE_ARG_VALUE(name, "name", name.c_str());
+
+        if (inputs_arr.depth() == CV_16F)
+        {
+            forward_fallback(inputs_arr, outputs_arr, internals_arr);
+            return;
+        }
+
+        std::vector<Mat> inputs, outputs;
+        inputs_arr.getMatVector(inputs);
+        outputs_arr.getMatVector(outputs);
+
+        Mat src = inputs[0];
+        Mat dst = outputs[0];
+
+        axis = normalize_axis(axis, src.dims);
+        MatShape shape(src.size.p, src.size.p + src.dims);
+
+        // Prepare output
+        memset(dst.ptr(), 0, dst.total() * dst.elemSize());
+
+        switch (src.depth())
+        {
+            case CV_8U: hardmaxApply<uchar>(src, dst, axis); break;
+            case CV_8S: hardmaxApply<schar>(src, dst, axis); break;
+            case CV_16U: hardmaxApply<ushort>(src, dst, axis); break;
+            case CV_16S: hardmaxApply<short>(src, dst, axis); break;
+            case CV_32S: hardmaxApply<int>(src, dst, axis); break;
+            case CV_32F: hardmaxApply<float>(src, dst, axis); break;
+            case CV_64F: hardmaxApply<double>(src, dst, axis); break;
+            default:
+                CV_Error(Error::StsUnsupportedFormat, "Unsupported input data type");
+        }
+    }
+
+    template <typename T>
+    void hardmaxApply(const cv::Mat& src, cv::Mat& dst, const int axis)
+    {
+        const auto *src_ptr = src.ptr<T>();
+        auto *dst_ptr = dst.ptr<T>();
+
+        const size_t outer_size = src.total(0, axis);
+        const auto mid_size = static_cast<size_t>(src.size[axis]);
+        const size_t inner_size = src.total(axis + 1);
+        const size_t outer_step = src.total(axis);
+
+        parallel_for_(Range(0, outer_size), [&](const Range& range) {
+            for (size_t outer = range.start; outer < range.end; ++outer)
+            {
+                const size_t outer_offset = outer * outer_step;
+
+                for (size_t inner = 0; inner < inner_size; ++inner)
+                {
+                    T max_val = std::numeric_limits<T>::lowest();
+                    size_t max_idx = 0;
+
+                    // Find max along the reduction axis
+                    for (size_t mid = 0; mid < mid_size; ++mid)
+                    {
+                        const size_t src_idx = outer_offset + mid * inner_size + inner;
+                        if (src_ptr[src_idx] > max_val)
+                        {
+                            max_val = src_ptr[src_idx];
+                            max_idx = src_idx;
+                        }
+                    }
+
+                    // Set 1 for max, 0 for others
+                    dst_ptr[max_idx] = 1;
+                }
+            }
+        });
```

Comment:
Add `nstripes` behind the lambda function. It equals to total number of loops divided by a constant, e.g. `outer_size * inner_size * (1 / 1024.0)`.

Reply:
where does `(1/1024.0)` or `1024.0` constant comes from? Just want to understand for myself.

Reply:
I learned it from other folks. They said that it is empirical. You could tune it if performance is critical.

My understanding is that I want each worker takes care of 1024 times of loops, and there I need `total_loops / 1024` workers in total from the thread pool.